### ISSUE TITLE
Model detects only 1 class

### ISSUE THREAD

**Issue by bassmaamn** [EdjeElectronics/TensorFlow-Object-Detection-API-Tutorial-Train-Multiple-Objects-Windows-10#7]
I want to train the model to detect 2 classes, but after training it recognize only 1 class

**Comment by sTANAN** [comment]
I have same issue. Did you find any solution ?

**Comment by KingsonSingh** [comment]
I trained two models with 14 classes and 6 classes, but after training both model it recognize only 1 class.

Did any one finds the solution?
Please share !!
Thank you !!!

**Comment by skvravi108** [comment]
I trained on 7 classes, But after training the model it is able to recognize only 1 class.

Can any one suggest to detect multiple classes.

**Comment by noelcodes** [comment]
I had the same problem. Trained 12 class, only recognize 6. Where 6 class is same number as Edje. I suspected there are more to modify. Please help. 
Below are codes I have changed.
faster_rcnn_inception_v2_pets,config  ==> num_classes: 12
labelmap.pbtxt ==> added 12 classes
generate_tfrecord.py ==> added 12 classes @ class_text_to_int
Object_detection_webcam.py ==> NUM_CLASSES = 12
Where else?

**Comment by leccyril** [comment]
you have to modify label.pbtxt file and model training config file and generate_tf.py file

**Comment by leccyril** [comment]
@JayLee15 did you modify the pipeline.config too ? there is the number class to specify too

**Comment by JayLee15** [comment]
@leccyril Yes, I have modified the number class in pipeline.config, but it recognizes only 1 class.

**Comment by leccyril** [comment]
what sample training did you use ? coco or pet detection ? maybe you can share your configuration files 

generate tf record, label.pbtxt and the model configuration file

2 solutions, either you have somewhere 1 class configured
or you have only few pictures for the second class

did you launch the train and eval and follow it on tensorboard ?

**Comment by mking1011** [comment]
@JayLee15 

Did you solve this problem?

**Comment by trc2019** [comment]
the same problem

**Comment by leccyril** [comment]
@mking1011 

it seems to be OK , what's happening when you check the eval on tensorboard ? 

@simonegrazioso 

it is what you see on eval board ?

**Comment by simonegrazioso** [comment]
> there is no problem to train with multi classes just fill files i've mentionned before
the files are all filled, otherwise there will be only one class detected and always the same one. Indeed, I detect both classes, rarely together, with very low accuracy when trained together. If i train the two classes separately, I have a good precision (50-55%). 

> @simonegrazioso
> 
> it is what you see on eval board ?

Exactly.

**Comment by Chloejay** [comment]
I have the same problem, the output just return one class, can someone advised me is I shouldn't write this in the code 
with tf.Session() as sess:
            for _image, _image_name in zip(_images,_image_names):
                _image_expanded = np.expand_dims(_image, axis=0) 
                image_tensor = tf.get_default_graph().get_tensor_by_name('image_tensor:0') 
                scores= g1.get_tensor_by_name('detection_scores:0') 
                classes= g1.get_tensor_by_name('detection_classes:0')     
                (scores, classes) = sess.run([scores, classes], 
                                             feed_dict={image_tensor:_image_expanded}) 
                for score, cls in zip(scores, classes):
                    score = score[0]
                    cls = cls[0] 
                    _results.append({"name":_image_name, "pred_prob":score, "class_name":cls}) 
                    print(score, cls) 
            
    return _results  

I take the last one as the model output

**Comment by niyanthh87** [comment]
In my case it was in label_map.pbtxt the label name i had given was "Car" where in annotation it was "car" check the annotation label and write accordingly as label_map.pbtxt is case Sensitive

**Comment by keertika-j** [comment]
While converting the files from xml to csv , did you mention all the labels properly ?

**Comment by drajsel** [comment]
According to this [https://stackoverflow.com/questions/54420363/multi-class-object-detection-with-tensorflow-weird-behavior-on-evaluation](answer), the filenames have to be shuffled inside the generate_tfrecords.py in order to properly train the model to detect multiple classes.

**Comment by Sangharsh-kamble** [comment]
you have to check labelmap file ie. .pbtxt files. it has id number for each label. also while creating .record file need to be assign this labels in code

**Comment by mehmetsoylu** [comment]
@bassmaamn  i have the same problem with yours. Did you solved your problem? If  you solved it, how did you solved it? Can you please explain to me?

**Comment by JayLee15** [comment]
> @JayLee15 @bassmaamn @leccyril @bassmaamn i have the same problem with yours. Did you solved your problem? If you solved it, how did you solved it? Can you please explain to me?
In my case, all categories of images are very similar. I increase the parameter 'second_stage_classification_loss_weight' and retrain to solve this problem.

**Comment by mehmetsoylu** [comment]
@JayLee15 i am new on object detection so i dont know how to do that. How do i increase my 'second_stage_classification_loss_weight' ?

**Comment by SOTASHE** [comment]
@mehmetsoylu  I'm facing the same problem..  I have 5 classes but my model only detect the first class

**Comment by devil1209** [comment]
i trained the object detection model for 13 classes using ssd mobilenetv2 but it is always predicting one class which has the lowest number of images  and i have also checked the label map and annnotations files. can anyone could suggest something here
